### DISCUSSION TITLE
emojicoin.fun doesn't work with revela

### DISCUSSION THREAD

**Discussion by MoonShiesty** [aptos-labs/aptos-developer-discussions#576] [Questions]
### Discord user ID

moon_shiesty

### Describe your question in detail.

emojicoin.fun contracts at `0xface729284ae5729100b3a9ad7f7cc025ea09739cd6e7252aff0beb53619cafe` cannot be decompiled with the online revela, i haven't tried using the cli

<img width="1124" alt="image" src="https://github.com/user-attachments/assets/aa9062cf-9190-4a72-8095-0e7f3b680721" />

### What error, if any, are you getting?

decompiled failed

### Describe your environment or tooling in detail

revela, chrome, aptos mainnet

**Comment by wrwg** [comment]
Revela does not support Move 2 bytecode, and will not be extended for Move 2, instead a new decompiler is in the make, but takes a bit longer to be ready.

When you run the decompiler from the newest CLI, the bytecode will be downgraded to pre-Move 2, which works provided no actual features like enum types are used. The online version (which we not maintain) doesn't do this.
